beacon_node/eth2-libp2p/src/rpc/handler.rs

```diff
@@ -806,75 +819,112 @@ where
                         OutboundSubstreamState::RequestPendingResponse {
                             mut substream,
                             request,
-                        } => match substream.poll_next_unpin(cx) {
-                            Poll::Ready(Some(Ok(response))) => {
-                                if request.multiple_responses() && !response.is_error() {
+                        } => {
+                            match substream.poll_next_unpin(cx) {
+                                Poll::Ready(Some(Ok(response))) => {
+                                    if request.multiple_responses() && !response.is_error() {
+                                        let substream_entry = entry.get_mut();
+
+                                        let this_request_with_a_pending_reponse =
+                                            OutboundSubstreamState::RequestPendingResponse {
+                                                substream,
+                                                request,
+                                            };
+
+                                        substream_entry.0 = this_request_with_a_pending_reponse;
+                                        let delay_key = &substream_entry.1;
+                                        let remaining_number_of_chunks_for_this_stream =
+                                            substream_entry.3;
+                                        let new_amount_of_remaning_chunks_after_the_current_one =
+                                            remaining_number_of_chunks_for_this_stream
+                                                .map(|count| count.saturating_sub(1))
+                                                .unwrap_or_else(|| 0);
+                                        // close the stream if all expected chunks have been received
+                                        if new_amount_of_remaning_chunks_after_the_current_one == 0
+                                        {
+                                            match this_request_with_a_pending_reponse {
```

Comment:
I am having a compiler error right here.

The problem is that the the substream gets moved on line 830.
I need to access it again in this section, but got stuck. Any help is appreciated.

Reply:
For now I was able to solve this by moving 
```
   substream_entry.0 = OutboundSubstreamState::RequestPendingResponse {
                                                    substream,
                                                    request,
                                                };
```
into the else block that executes if not all response chunks have arrived yet.

I hope this is not a problem @AgeManning

Reply:
This looks fine to me